### PR TITLE
ignore "PR: Breaking Change" label when generating changelog

### PR DESCRIPTION
**Pull request by JLHwung**
| Q                        | A 
| ------------------------ | ---
| License                  | MIT

After we release Babel 8, I expect the flag `BABEL_8_BREAKING` will simply evolve to `BABEL_9_BREAKING`,  but I am not sure we are going to create the PR label `babel 9` and `babel 9 -> main` again. In this PR I suggest we revive the `PR: Breaking Change :boom:` label, these PRs will be merged only if they are implemented after a breaking-major env flag. They will be ignored by our release changelog, as they are merged but never shipped.

If we want to generate breaking changes between major versions, we can rename the `labels_breaking` to `labels` and `labels` to `labels_disabled`, and then manually run `lerna-changelog` from the first version of this major to the next major, e.g.

```
node ./node_modules/.bin/lerna-changelog --tag-from 8.0.0 --tag-to 9.0.0
```

It should then generate a complete changelog including all changes shipped to 9.0.0.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/15848"></a>

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
I think there will be some non-breaking changes that will only land in Babel 8 (because, for example, they depend on a breaking change), but this is much better than the current config.

File: package.json
Comment (nicolo-ribaudo) [inline package.json:116]: Does lerna-changelog complain if we do this?
```js
"changelog": {
  ...,
  "labels": {
    the breaking label is not here
  },
  "labels_breaking": {
    "PR: Breaking Change :boom:": ":boom: Breaking Change"
  }
}
```

File: package.json
Comment (JLHwung) [inline package.json:127]: They can re-use the current `Babel 8` purple colorr, so it is less likely that maintainers are using incorrect labels.

File: package.json
Comment (JLHwung) [inline package.json:116]: Previously the `Deprecation` PR was never included in the change log because of the extra colon. This is now fixed.

**Comment by nicolo-ribaudo** [comment]
After that this is merged I'll update the Babel 8 release scripts to properly work with it.

**Comment by JLHwung** [comment]
> After that this is merged I'll update the Babel 8 release scripts to properly work with it.

Thanks. And I will search previous Babel 8 PRs and associate them with new labels.

File: package.json
Comment (JLHwung) [inline package.json:136]:
```suggestion
      "PR: Revert (next major)": ":leftwards_arrow_with_hook: Revert",
```
GitHub PR labels are limited to 50 characters.

**Comment by nicolo-ribaudo** [comment]
Wdyt about also cloring `PR: Breaking Change` purple rather than yellow?

**Comment by JLHwung** [comment]
> Wdyt about also cloring `PR: Breaking Change` purple rather than yellow?

I thought about it. If we are not going to introduce `BABEL_9_BREAKING` flag _parallel to_ `BABEL_8_BREAKING`, which is a breaking change with respective to the next major version, I am good to make it purple.

**Comment by liuxingbaoyu** [comment]
I have an idea, but I'm not sure if the tools support it.
When a PR has only `PR: Breaking Change`, then it will be included in the changelog.
If it has both `PR: Breaking Change` and `babel 8`, then it will be temporarily ignored.

**Comment by nicolo-ribaudo** [comment]
@liuxingbaoyu That's much harder because it requires modifying lerna-changelog. This approach is easier because we just need to use the correct labels :)

**Comment by liuxingbaoyu** [comment]
I get it, the current approach already looks a lot better than it used to.

Do we still need to use `babel 8` at the same time?
I worry that in a few years someone will want to check them out.

**Comment by JLHwung** [comment]
> I have an idea, but I'm not sure if the tools support it. When a PR has only `PR: Breaking Change`, then it will be included in the changelog. If it has both `PR: Breaking Change` and `babel 8`, then it will be temporarily ignored.

And we are not going to, at least intentionally, ship breaking changes to Babel 7 as we follow the semantic versioning. That means a breaking change PR without an env toggle should not be merged at all.

**Comment by liuxingbaoyu** [comment]
> And we are not going to, at least intentionally, ship breaking changes to Babel 7 as we follow the semantic versioning. That means a breaking change PR without an env toggle should not be merged at all.

Yeah. I just noticed that all tags have a `(next major)` version. :)

**Comment by JLHwung** [comment]
> I get it, the current approach already looks a lot better than it used to.
> 
> Do we still need to use `babel 8` at the same time? I worry that in a few years someone will want to check them out.

I tend to use the Babel 8.0 milestone to group such PRs, people can check PR's included in every minor: https://github.com/babel/babel/milestones?state=closed

**Review by nicolo-ribaudo — APPROVED** [review]

**Review by liuxingbaoyu — APPROVED** [review]

**Comment by nicolo-ribaudo** [comment]
I'm experimenting locally with the GH workflow updates

**Pull request merged**
